akka-stream-tests/src/test/scala/akka/stream/impl/fusing/ActorGraphInterpreterSpec.scala

```diff
@@ -331,5 +335,64 @@ class ActorGraphInterpreterSpec extends AkkaSpec {
 
     }
 
+    "be able to handle Publisher spec violations without leaking" in assertAllStagesStopped {
+      val filthyPublisher = new Publisher[Int] {
+        override def subscribe(s: Subscriber[_ >: Int]): Unit = {
+          s.onSubscribe(new Subscription {
+            override def cancel(): Unit = ()
+            override def request(n: Long): Unit = throw TE("violating your spec")
+          })
+        }
+      }
+
+      val upstream = TestPublisher.probe[Int]()
+      val downstream = TestSubscriber.probe[Int]()
+
+      Source.combine(
+        Source.fromPublisher(filthyPublisher),
+        Source.fromPublisher(upstream))(count ⇒ Merge(count))
+        .runWith(Sink.fromSubscriber(downstream))
+
+      upstream.ensureSubscription()
+      upstream.expectCancellation()
+
+      downstream.ensureSubscription()
+      downstream.expectError() match {
+        case ise: IllegalStateException ⇒
```

Comment:
never mind, I refactored it a bit. Also, for me `and have` did not work, neither `should be`, but `shouldBe` did. So I deviated somewhat from your suggestion in the end.